Client/Assets/LoginsHelper.js

```diff
@@ -599,19 +599,28 @@ function onBlur(event) {
   LoginManagerContent.onUsernameInput(event)
 }
 
-window.addEventListener("load", function(event) {
+var documentBody = document.querySelector('body')
+var observer = new MutationObserver(function(mutations) {
+    findLogins()
+});
+
+observer.observe(documentBody, { attributes: false, childList: true, characterData: false });
+
+function findLogins(event) {
   try {
     for (var i = 0; i < document.forms.length; i++) {
       LoginManagerContent._asyncFindLogins(document.forms[i], { })
```

Comment:
By looping over all forms on the page, it looks like you'll end up re-triggering the query for all forms every time a new one is added. Instead, you can use the `MutationObserver` to filter out `form`-specific additions, and ignore other events. For example:

```
var observer = new MutationObserver(function(mutations) {
  for (var i = 0; i < mutations.addedNodes.length; i++) {
    var node = mutations.addedNodes[i];
    if (node.tagName === "FORM") {
      LoginManagerContent._asyncFindLogins(node, { });
    }
  }
});
```